Platform/AlderlakeBoardPkg/Library/Stage1BBoardInitLib/EcSupport.c

```diff
@@ -246,9 +311,39 @@ GetBoardId (
   }
   BoardInfo.Raw |= (UINT16) EcData;
 
-  *PlatformId = (UINT8) BoardInfo.AdlRvpFields.BoardId;
+  *BoardId = (UINT8)BoardInfo.AdlRvpFields.BoardId;
+}
+
+/**
+  Get the board ID from EC.
+
+  @param[out] PlatformId       Buffer to return the Platform ID based on EC BaordId
 
-  switch (BoardInfo.AdlRvpFields.BoardId) {
+**/
+VOID
+GetBoardId (
+  OUT UINT8                  *PlatformId
+)
+{
+  UINT8   BoardID;
+
+  //
+  // Set board ID to unknown initially
+  // in case there is any issue encountered
+  // below.
+  //
+  *PlatformId = 0xFF;
+
+#ifdef PLATFORM_ADLN
```

Comment:
If both EC and SMBus takes time to read BoardId, we might add a config data to indicate which one to use so that we could share same SBL.

Reply:
is there any config data that can be use before the platform id was initialized?